packages/core/scripts/local/GasEthQuery.js

```diff
@@ -0,0 +1,62 @@
+// This is a helper method to create a GASETH BQ query with time arguments.
+function createQuery(earlierTimeBound, laterTimeBound) {
+  const query = `
+  DECLARE halfway int64;
+  DECLARE block_count int64;
+  DECLARE max_block int64;
+
+  -- Querying for the amount of blocks in the preset time range. This will allow block_count to be compared against a given minimum block amount.
+  SET (block_count, max_block) = (SELECT AS STRUCT (MAX(number) - MIN(number)), MAX(number) FROM \`bigquery-public-data.crypto_ethereum.blocks\` 
+  WHERE timestamp BETWEEN TIMESTAMP('${earlierTimeBound}', 'UTC') AND TIMESTAMP('${laterTimeBound}', 'UTC'));
+
+  CREATE TEMP TABLE cum_gas (
+    gas_price int64,
+    cum_sum int64
+  );
+
+  -- If the minimum threshold of blocks is met, query on a time range
```

Comment:
this is the same bigquery we had back when you first designed the uGAS feeds, right? before moving over to the twap pattern?

Reply:
Yes it is the same.